src/psyclone/domain/lfric/transformations/lfric_invoke_trans.py

```diff
@@ -0,0 +1,184 @@
+        super(LFRicInvokeTrans, self).validate(call, options=options)
+
+    def _validate_fp2_node(self, fp2_node):
+        '''Specialisation of the fparser2 node validation routine to
+        additionally validate named arguments, which are specific to
+        the LFRic API.
+
+        :param fp2_node: an fparser2 Structure Constructor or Actual \
+            Arg Spec node.
+        :type fp2_node: \
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor` or \
+            :py:class:`fparser.two.Fortran2003.Actual_Arg_Spec
+
+        :raises TransformationError: if the named argument is not in \
+            the expected form.
+        :raises TransformationError: if more than one named argument \
+            is found.
+        :raises TransformationError: if the fparser2 node is not the \
+            expected type.
+
+        '''
+        if isinstance(fp2_node, Structure_Constructor):
+            pass
+        elif isinstance(fp2_node, Actual_Arg_Spec):
+            if not (isinstance(fp2_node.children[0], Name) and
+                    fp2_node.children[0].string.lower() == "name" and
+                    isinstance(fp2_node.children[1], Char_Literal_Constant)):
+                raise TransformationError(
+                    "Error in {0} transformation. If there is a named "
+                    "argument, it must take the form name='str', but found "
+                    "'{1}'.".format(self.name, str(fp2_node)))
+            if self._call_description:
+                raise TransformationError(
+                    "Error in {0} transformation. There should be at most one "
+                    "named argument in an invoke, but there are at least two: "
+                    "{1} and {2}.".format(self.name, self._call_description,
+                                          fp2_node.children[1].string))
+            self._call_description = fp2_node.children[1].string
+        else:
+            raise TransformationError(
+                "Error in {0} transformation. Expecting an algorithm invoke "
+                "codeblock to contain either Structure-Constructor or "
+                "actual-arg-spec, but found '{1}'."
+                "".format(self.name, type(fp2_node).__name__))
+
+    def apply(self, call, options=None):
+        ''' Apply the transformation to the supplied node.
+
+        :param call: a PSyIR call node capturing an invoke call in \
+            generic PSyIR.
+        :type call: :py:class:`psyclone.psyir.nodes.Call`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        '''
+        self.validate(call, options=options)
+
+        call_description = None
+        calls = []
+        for call_arg in call.children:
+
+            arg_info = []
+            if isinstance(call_arg, ArrayReference):
+                # kernel or builtin misrepresented as ArrayReference
+                args = call_arg.children
```

Comment:
If you detach the children before appending them on the arg_info list you way not need the `args = [arg.copy() for arg in args]` below

Reply:
Yes this works. Thanks.